src/pygeometadata/mcf.py

```diff
@@ -0,0 +1,413 @@
+        Args:
+            name (str): name and unique identifier of the field
+            title (str): title for the field
+            abstract (str): description of the field
+            units (str): unit of measurement for the field's values
+        """
+        def get_attr(attribute_list):
+            for idx, attr in enumerate(attribute_list):
+                if attr['name'] == name:
+                    return idx, attr
+            raise ValueError(
+                f'{self.datasource} has no attribute named {name}')
+
+        idx, attribute = get_attr(self.mcf['content_info']['attributes'])
+
+        if title is not None:
+            attribute['title'] = title
+        if abstract is not None:
+            attribute['abstract'] = abstract
+        if units is not None:
+            attribute['units'] = units
+
+        self.mcf['content_info']['attributes'].insert(idx, attribute)
+
+    def write(self):
+        """Write MCF to disk."""
+        with open(self.mcf_path, 'w') as file:
+            file.write(yaml.dump(self.mcf))
+        # TODO: always also write an ISO-191* XML doc?
+
+    def validate(self):
+        """Validate MCF against a jsonschema object."""
+        # validate against our own schema, which could
+        # be a superset of the core MCF schema.
+        # If we wanted to validate against core MCF,
+        # we could use pygeometa.core.validate_mcf
+        jsonschema.validate(self.mcf, MCF_SCHEMA)
+
+    def to_string(self):
+        pass
+
+    def get_spatial_info(self):
+        """Populate the MCF using properties of the dataset."""
+        gis_type = pygeoprocessing.get_gis_type(self.datasource)
+        if gis_type == pygeoprocessing.UNKNOWN_TYPE:
+            self.mcf['metadata']['hierarchylevel'] = 'nonGeographicDataset'
+            return
+
+        if gis_type == pygeoprocessing.VECTOR_TYPE:
+            self.mcf['metadata']['hierarchylevel'] = 'dataset'
+            self.mcf['spatial']['datatype'] = 'vector'
+            self.mcf['content_info']['type'] = 'coverage'
+
+            vector = gdal.OpenEx(self.datasource, gdal.OF_VECTOR)
+            layer = vector.GetLayer()
+            layer_defn = layer.GetLayerDefn()
+            geomname = ogr.GeometryTypeToName(layer_defn.GetGeomType())
+            geomtype = ''
+            # https://www.fgdc.gov/nap/metadata/register/codelists.html
+            if 'Point' in geomname:
+                geomtype = 'point'
+            if 'Polygon' in geomname:
+                geomtype = 'surface'
+            if 'Line' in geomname:
+                geomtype = 'curve'
+            if 'Collection' in geomname:
+                geomtype = 'complex'
+            self.mcf['spatial']['geomtype'] = geomtype
+
+            attributes = []
+            for field in layer.schema:
+                attribute = {}
+                attribute['name'] = field.name
+                attribute['type'] = OGR_MCF_ATTR_TYPE_MAP[field.type]
```

Comment:
Here's the schema for `attribute.type`:

```yaml
      type:
          type: string
          description: data type of attribute
          enum:
              - string
              - number
              - integer
              - object
              - array
              - boolean
```
`enum` allows for only one of those values. So I'm kind of leaning towards deleting the `attribute['type']` in that case -- it's not required by the schema for `attribute` -- rather than choosing an arbitrary value that might be incorrect.

Reply:
I think that sounds like a good solution ... maybe we could issue a warning or something if we hit that case?

Reply:
Patched in https://github.com/natcap/pygeometadata/pull/2/commits/9e78e012972017daa63b91249a57958e1ce5c90e